src/DesktopHeader.jsx

```diff
@@ -74,21 +74,25 @@ class DesktopHeader extends React.Component {
 
   renderUserMenu() {
     const {
+      intl,
       userMenu,
       avatar,
+      name,
       username,
-      intl,
     } = this.props;
+    const hideUsername = getConfig().HIDE_USERNAME_IN_HEADER;
+    const usernameOrName = hideUsername ? name : username;
 
     return (
       <Menu transitionClassName="menu-dropdown" transitionTimeout={250}>
         <MenuTrigger
           tag="button"
-          aria-label={intl.formatMessage(messages['header.label.account.menu.for'], { username })}
+          aria-label={intl.formatMessage(messages['header.label.account.menu.using.name.or.username'], { usernameOrName })}
           className="btn btn-outline-primary d-inline-flex align-items-center pl-2 pr-3"
         >
           <Avatar size="1.5em" src={avatar} alt="" className="mr-2" />
-          {username} <CaretIcon role="img" aria-hidden focusable="false" />
+          {hideUsername ? null : username}
```

Comment:
nit: 
```suggestion
          {!hideUsername && username}
```

Reply:
We don't want to use negative condition because it may hard to understand sementic.

Reply:
it doesn't seem like very complex to me.

Reply:
The comment was to not negate the config. You can negate the coditions.